### PR TITLE
Add Nginx to tests

### PR DESCRIPTION
**Pull request by BrennanConroy**
@JunTaoLuo @Tratcher

### CODE REVIEW

File: test/FunctionalTests/Diagnostics/DiagnosticsStatusCodesMvcTests.cs [hunk 1]
```diff
@@ -38,6 +38,7 @@ public class DiagnosticsStatusCodesMvcTests
         [OSSkipCondition(OperatingSystems.Windows)]
         [InlineData(ServerType.Kestrel, RuntimeFlavor.Clr, RuntimeArchitecture.x64, "http://localhost:6109")]
         [InlineData(ServerType.Kestrel, RuntimeFlavor.CoreClr, RuntimeArchitecture.x64, "http://localhost:6110")]
+        [InlineData(ServerType.Nginx, RuntimeFlavor.CoreClr, RuntimeArchitecture.x64, "http://localhost:6111")]
```
Comment (JunTaoLuo) [inline test/FunctionalTests/Diagnostics/DiagnosticsStatusCodesMvcTests.cs]: Fix the port conflicts between this test and the other ones. Might want to check you other changes too.

**Comment by JunTaoLuo** [comment]
We need a better way of assigning ports... adding tests are a pain since all the port numbers need to change.

**Comment by BrennanConroy** [comment]
:up: 📅

File: .travis.yml [hunk 2]
```diff
@@ -16,13 +16,14 @@ os:
   - linux
   - osx
 osx_image: xcode7.1
+before_install:
+  - if test "$TRAVIS_OS_NAME" == "osx"; then brew update; brew install nginx; fi
 install:
-  - curl -sSL https://github.com/libuv/libuv/archive/v1.4.2.tar.gz | tar zxfv - -C /tmp && cd /tmp/libuv-1.4.2/
-  - sh autogen.sh
-  - ./configure --prefix=$HOME/libuvinstall
+  - curl -sSL http://nginx.org/download/nginx-1.8.0.tar.gz | tar zxfv - -C /tmp && cd /tmp/nginx-1.8.0/
```
Comment (anurse) [inline .travis.yml]: Is this also going to run on OSX? If so, why are we `brew install`ing nginx and then building it from source?

Comment (BrennanConroy) [inline .travis.yml]: Because reasons. I'll change it

**Comment by anurse** [comment]
LGTM, :shipit: after resolving comments and getting green build.

**Comment by anurse** [comment]
Also, this is super old. Are we taking it for RTM?

**Comment by BrennanConroy** [comment]
No reason to take it for RTM, I'll do it after we branch

**Pull request merged**